### DISCUSSION TITLE
What is IPython needed for?

### DISCUSSION THREAD

**Discussion by humcasma** [vaexio/vaex#1415] [Q&A]
Vaex complains if running on a notebook with a version of IPython smaller than 7.0.0. I guess this check exists because some parts of vaex, such _vaex-jupyter_, may need an updated IPython library. This makes running Vaex in Google Colab problematic, since Google Colab comes with IPython 5.5.0 and may crash and behave erratically when upgrading to a newer version.

Right now I am just using _vaex-core_ and _vaex-hdf5_ in Google Colab, so I tried disabling the IPython check and everything seems to work fine, without upgrading IPython. I would like to know if this is safe or whether the data/results could be silently getting corrupted. In summary, what parts of Vaex do really need an updated IPython library and why?

**Comment by maartenbreddels** [comment]
When running in IPython, there is an asyncio loop installed. This can cause issues with some asyncio parts in Vaex. However, if IPython is not used, it does not matter.
Hope that answers your question.

**Reply by humcasma** [reply]
Thanks @maartenbreddels 
Just to be sure, is there any specific functionality in IPython 7.0.0, not present in version 5.5.0, that Vaex requires? Or should Vaex work equally well with version 5.5.0?
If using version 5.5.0 is problematic, what problems can arise? Instabilities? A crash? A data corruption (e.g., that a groupby is not properly performed)?

**Comment by maartenbreddels** [comment]
If you only have ipython installed, but not using it, it's not a problem.
Otherwise I'm the notebook, the cell output might be messed up, or go to
the wrong cell

(from mobile phone)
